.github/workflows/image.yaml

```diff
@@ -0,0 +1,56 @@
+name: Image
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    env:
+      GOPATH: /home/runner/work/argo-cd/argo-cd
+    steps:
+      - uses: actions/setup-go@v1
+        with:
+          go-version: '1.12.1'
+      - uses: actions/checkout@master
+        with:
+          path: src/github.com/argoproj/argo-cd
+      - uses: actions/cache@v1
+        with:
+          path: src/github.com/argoproj/argo-cd/vendor
+          key: ${{ runner.os }}-go-dep-${{ hashFiles('**/Gopkg.lock') }}
+      # download dependencies
+      - run: mkdir -p $GITHUB_WORKSPACE/bin && curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+        working-directory: src/github.com/argoproj/argo-cd
+      - run: $GOPATH/bin/dep ensure -v
+        working-directory: ./src/github.com/argoproj/argo-cd
+
+      # get image tag
+      - run: echo ::set-output name=tag::$(cat ./VERSION)-${GITHUB_SHA::8}
```

Comment:
I wonder how many versions/images will retain? Will we have cleanup jobs, or is the retention configurable? Generally, I like the idea of a unique version identifier for each commit vs a regular `latest`, so that each commit can be tested isolated.

Reply:
Unfortunately, Github does not support delete old images yet. I hope that functionality will be added soon and we can configure the retention policy/delete old images in the CI.

In the worst case, we can switch to separate org in dockerhub.